modules/simple-payments/simple-payments.php

```diff
@@ -0,0 +1,225 @@
+<?php
+/*
+ * Simple Payments lets users embed a PayPal button fully integrated with wpcom to sell products on the site.
+ * This is not a proper module yet, because not all the pieces are in place. Until everything is shipped, it can be turned
+ * into module that can be enabled/disabled.
+ * TODO: Once the feature is fully shipped, create a file modules/simple-payments.php with a proper header to turn module on/off.
+*/
+class Jetpack_Simple_Payments {
+	// These have to be under 20 chars because that is CPT limit.
+	static $post_type_order = 'jp_pay_order';
+	static $post_type_product = 'jp_pay_product';
+
+	static $shortcode = 'simple-payment';
+
+	// Classic singleton pattern:
+	private static $instance;
+	private function __construct() {}
+	static function getInstance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+			self::$instance->register_init_hook();
+		}
+		return self::$instance;
+	}
+
+	private function register_scripts() {
+		/**
+		 * Paypal heavily discourages putting that script in your own server:
+		 * @see https://developer.paypal.com/docs/integration/direct/express-checkout/integration-jsv4/add-paypal-button/
+		 */
+		wp_register_script( 'paypal-checkout-js', 'https://www.paypalobjects.com/api/checkout.js' );
+		wp_register_script( 'paypal-express-checkout', plugins_url( '/paypal-express-checkout.js', __FILE__ ) , array( 'paypal-checkout-js' ), '0.21' );
+	}
+	private function register_init_hook() {
+		add_action( 'init', array( $this, 'init_hook_action' ) );
+	}
+	private function register_shortcode() {
+		add_shortcode( self::$shortcode, array( $this, 'parse_shortcode' ) );
+	}
+
+	public function init_hook_action() {
+		add_filter( 'rest_api_allowed_post_types', array( $this, 'allow_rest_api_types' ) );
+		add_filter( 'jetpack_sync_post_meta_whitelist', array( $this, 'allow_sync_post_meta' ) );
+		$this->register_scripts();
+		$this->register_shortcode();
+		$this->setup_cpts();
+	}
+
+	function parse_shortcode( $attrs, $content = false ) {
+		if( empty( $attrs['id'] ) ) {
+			return;
+		}
+		$product = get_post( $attrs['id'] );
+		if( is_wp_error( $product ) ) {
+			return;
+		}
+		if( $product->post_type !== self::$post_type_product ) {
+			return;
+		}
+
+		// We allow for overriding the presentation labels
+		$data = shortcode_atts( array(
+			'blog_id' => $blog_id = Jetpack_Options::get_option( 'id' ),
+			'dom_id' => uniqid( 'jetpack-simple-payments-' . $product->ID . '_' ),
+			'class' => 'jetpack-simple-payments-' . $product->ID,
+			'title' => get_the_title( $product ),
+			'description' => apply_filters( 'the_content', $product->post_content ),
```

Comment:
We are dropping that filter since we don't want to have HTML tags in `description` field.